### PR TITLE
Sequence cluster addons creation

### LINKED ISSUE #12: vpc-cni addon configuration example
### What problem are you facing?

As a follow-up to https://github.com/upbound/configuration-aws-eks/pull/11

we need to provide example of vpc-cni addon configuration, similar to

```

      - name: vpcCniAddon
        base:
          apiVersion: eks.aws.upbound.io/v1beta1
          kind: Addon
          spec:
            forProvider:
              addonName: vpc-cni
              resolveConflicts: OVERWRITE
              clusterNameSelector:
                matchControllerRef: true
              configurationValues: '{"env": {"AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG":"true", "ENI_CONFIG_LABEL_DEF":"topology.kubernetes.io/zone", "WARM_IP_TARGET":"10", "WARM_ENI_TARGET":"0", "WARM_PREFIX_TARGET": "0"}}'
```

As was pointed out by @haarchri , it should also involve the `vpc-cni` creation ordering, as addon should be created after the Cluster but before NodeGroup for proper config propagation. 

### How could Upbound help solve your problem?

Extend Configuration with the associated example

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* Desired sequence Cluster -> CNIAddon -> NodeGroup -> ebsCsiAddon
* Use function-sequencer for reliable sequenced creation
* Add non-intrusive configuration example for vpc-cni
* Impementation for both standard P&T and KCL functions
* Fixes #12

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

`make e2e`

Observed with `crossplane beta trace` that associated resources are sequenced properly during the XEKS instantiation.

Double-check with resource age timestamps.
```
k get cluster,addons,nodegroups
NAME                                                                      SYNCED   READY   EXTERNAL-NAME                                  AGE
cluster.eks.aws.upbound.io/configuration-aws-eks-7qr2g                    True     True    configuration-aws-eks-7qr2g                    27m
cluster.eks.aws.upbound.io/configuration-aws-eks-kcl-kubernetes-cluster   True     True    configuration-aws-eks-kcl-kubernetes-cluster   27m

NAME                                                               SYNCED   READY   EXTERNAL-NAME                                                     AGE
addon.eks.aws.upbound.io/configuration-aws-eks-kcl-cni-addon       True     True    configuration-aws-eks-kcl-kubernetes-cluster:vpc-cni              13m
addon.eks.aws.upbound.io/configuration-aws-eks-kcl-ebs-csi-addon   True     True    configuration-aws-eks-kcl-kubernetes-cluster:aws-ebs-csi-driver   10m
addon.eks.aws.upbound.io/configuration-aws-eks-n9bv8               True     True    configuration-aws-eks-7qr2g:vpc-cni                               16m
addon.eks.aws.upbound.io/configuration-aws-eks-xf5r8               True     True    configuration-aws-eks-7qr2g:aws-ebs-csi-driver                    13m

NAME                                                                      SYNCED   READY   EXTERNAL-NAME                                AGE
nodegroup.eks.aws.upbound.io/configuration-aws-eks-kcl-nodegroup-public   True     True    configuration-aws-eks-kcl-nodegroup-public   12m
nodegroup.eks.aws.upbound.io/configuration-aws-eks-vmwhl                  True     True    configuration-aws-eks-vmwhl                  15m
```

### CODE REVIEW

**Review by stevendborrelli — APPROVED** [review]
One minor change needed.

File: examples/functions.yaml [hunk 1]
```diff
@@ -18,5 +18,11 @@ metadata:
   name: crossplane-contrib-function-auto-ready
 spec:
   package: xpkg.upbound.io/crossplane-contrib/function-auto-ready:v0.2.1
-
+---
+apiVersion: pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-sequencer
```
[the remark this answers is not in the document]
Comment (ytsarev) [reply examples/functions.yaml]: good catch! Done

**Review by haarchri — APPROVED** [review]

**Pull request merged**